File: Data Structures & Algorithms/max-water-container/submission-2.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l, r = 0, len(heights)-1
        max_area = 0

        while l < r:
            leftHeight = heights[l]
            rightHeight = heights[r]
            area = min(leftHeight, rightHeight) * (r-l)
            if leftHeight > rightHeight:
                r-=1
            else:
                l +=1
            max_area = max(area,max_area)
        
        return max_area

File: Data Structures & Algorithms/replace-elements-with-greatest-element-on-right-side/submission-2.cpp
class Solution {
   public:
    vector<int> replaceElements(vector<int>& arr) {
        int r = arr.size() - 1;
        int l = r - 1;
        int maxRight = arr[r];
        while (l >= 0) {
            maxRight = max(arr[l], arr[r]);
            arr[l] = arr[r];
            l--;
            arr[r] = maxRight;
        }
        arr[r] = -1;
        return arr;
    }
};